### PR TITLE
Benchpress: add receivedData + requestCount to PerflogMetric

### PR DESCRIPTION
**Pull request by ochafik**
Hey @tbosch, here's a timid attempt to add these measures :-)

### CODE REVIEW

**Comment by tbosch** [comment]
Wow, I am impressed. This looks really great!

File: modules/benchpress/src/webdriver/chrome_driver_extension.ts [hunk 1]
```diff
@@ -193,6 +193,15 @@ export class ChromeDriverExtension extends WebDriverExtension {
...
                this._isEvent(categories, name, ['devtools.timeline'], 'Paint')) {
       return normalizeEvent(event, {'name': 'render'});
+    } else if (this._isEvent(categories, name, ['devtools.timeline'], 'ResourceReceivedData')) {
+      let normArgs = {'encodedDataLength': args['data']['encodedDataLength']};
+      return normalizeEvent(event, {'name': 'receivedData', 'args': normArgs});
+    } else if (this._isEvent(categories, name, ['devtools.timeline'], 'ResourceSendRequest')) {
+      let data = args['data'];
+      let normArgs = {'url': data['url'], 'method': data['requestMethod']};
+      return normalizeEvent(event, {'name': 'sendRequest', 'args': normArgs});
+    } else if (this._isEvent(categories, name, ['blink.user_timing'], 'navigationStart')) {
```
Comment (tbosch) [inline modules/benchpress/src/webdriver/chrome_driver_extension.ts]: Could you also add the category `blink.user_timing` to https://github.com/angular/angular/blob/master/protractor-shared.js#L41? Or is it automatically included via the categories `v8,blink.console,devtools.timeline` that we already have there?

**Comment by tbosch** [comment]
Could you also add some docs about how to use this to https://github.com/angular/angular/blob/master/modules/benchpress/docs/index.md#smoothness-metrics? Similar to the smoothness-metrics?

**Comment by tbosch** [comment]
Please ping me directly when changes are done and Travis is green...

File: modules/benchpress/src/webdriver/chrome_driver_extension.ts (hunk 1 above)
Comment (ochafik) [inline modules/benchpress/src/webdriver/chrome_driver_extension.ts]: Seems to come for free through blink.console, actually.

**Comment by tbosch** [comment]
Sorry, I forgot: Could you use these features in https://github.com/angular/angular/blob/master/modules/playground/test/benchpress/webdriver_async.server.spec.dart, so we can can make sure they actually work?
I.e. create a new test that contains `await driver.get(...` inside of `runner.sample` call...

**Comment by tbosch** [comment]
Decided not to add an e2e test as some of the trace info is not available in Chrome 39 (Dartium), but our CI only runs Dartium right now.

**Comment by ochafik** [comment]
Thanks guys!

**Pull request closed without merging**